api/handler.go

```diff
@@ -301,11 +301,21 @@ func (h *apiHandler) queryUsage() http.HandlerFunc {
 		}
 
 		userId, ok := r.Context().Value(userIdContextKey).(string)
+		isCallerAdmin, _ := r.Context().Value(isCallerAdminContextKey).(string)
+
 		if !ok {
 			respondError(rw, http.StatusInternalServerError, errors.New("request not authenticated"))
 			return
 		}
 
+		if qs := r.URL.Query(); qs.Get("userId") != "" {
+			if isCallerAdmin == "false" {
+				respondError(rw, http.StatusForbidden, errors.New("only admins can query usage for other users"))
+				return
+			}
+			userId = qs.Get("userId")
+		}
+
 		qs := r.URL.Query()
```

Comment:
**nit:** This is being redefined as an if. Can probably just move this line up and avoid the repetition.